libraries/chain/controller.cpp

```diff
@@ -4058,28 +4040,79 @@ struct controller_impl {
       }
    }
 
-   // thread safe, expected to be called from thread other than the main thread
-   template<typename ForkDB, typename BS>
-   block_handle create_block_state_i( ForkDB& forkdb, const block_id_type& id, const signed_block_ptr& b, const BS& prev ) {
+   // verify basic_block invariants
+   template<typename BS>
+   void verify_basic_block_invariants(const signed_block_ptr& b, const BS& prev) {
```

Comment:
I don't think we need `basic` in the names `verify_basic_block_invariants` and `verify_basic_proper_block_invariants`.

Reply:
I think the `basic` here indicates we have more to check (like full qc) if required.